src/terminal/kitty.rs

```diff
@@ -230,18 +226,18 @@ impl KittyImages {
             _ => KittyFormat::RGBA,
         };
 
-        let (image_width, image_height) = image.dimensions();
-
-        let image = if image_width > terminal_size.width || image_height > terminal_size.height {
+        let image = if terminal_size.ge(image.dimensions()) {
+            image
+        } else {
             image.resize_to_fill(
                 terminal_size.width,
                 terminal_size.height,
                 FilterType::Nearest,
             )
-        } else {
-            image
         };
 
+        let (image_width, image_height) = image.dimensions();
```

Comment:
I don't think we need these names… I think you can inline them as `image.width()` and `image.height()` below.

Reply:
This is necessary because the variable image is already moved.

Reply:
Oh, yeah, ownership :)